### PR TITLE
#422 Display title in Agenda started

### PR DESCRIPTION
**Pull request by amihaiemil**
For #422 add_to_agenda.groovy now adds the title as well.

Modified the calls to Agenda.add(...) to also pass the title.

Could not yet add the title to the XML, inside Agenda.add(...), because the xsd validation fails.

Left puzzle to modify ``agenda.xsd`` from ``zerocracy/datum`` and to display the title on the Agenda page.

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#784](https://github.com/zerocracy/farm/issues/784) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#784](https://github.com/zerocracy/farm/issues/784) is assigned to @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda), here is [why](https://www.0crat.com/footprint/C3NDPUA8L/681760539). The budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4). Please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @yegor256[/z](https://www.0crat.com/u/yegor256) (the architect) right in this ticket. If you decide that this PR should not be accepted ever, also inform the architect.

File: src/main/java/com/zerocracy/tk/profile/TkAgenda.java [hunk 1]
```diff
@@ -31,6 +31,11 @@
  * @author Yegor Bugayenko (yegor256@gmail.com)
  * @version $Id$
  * @since 0.12
+ * @todo #422:30min Display the title of the order on the Agenda page.
+ *  The Issue's title is read from Github in add_to_agenda.groovy and
+ *  in the agenda, it is an attribute of element order, same as its id (job).
```
Comment (carlosmiranda) [inline src/main/java/com/zerocracy/tk/profile/TkAgenda.java]: @amihaiemil let's consolidate it into one puzzles. One is dependent on the other so if they get assigned simultaneously one of them will be blocked. Let ticket performer create the downstream puzzle for updating agenda page.

**Comment by amihaiemil** [comment]
@carlosmiranda what about this one?

File: src/main/java/com/zerocracy/pmo/Agenda.java [hunk 2]
```diff
@@ -33,6 +33,11 @@
  * @author Yegor Bugayenko (yegor256@gmail.com)
  * @version $Id$
  * @since 0.12
+ * @todo #422:30min Modify agenda.xsd zerocracy/datum and make the element
```
Comment (carlosmiranda) [inline src/main/java/com/zerocracy/pmo/Agenda.java]: @amihaiemil I'm inclined to reject this PR at the moment. Why go through the trouble of making all the changes in the code base while we can't actually make the single change that would make this ticket matter? Would be better to start the changes in `agenda.xsd` in datum and just have a ticket here without any further code change.

**Comment by carlosmiranda** [comment]
@amihaiemil one comment above. Sorry, I missed this somehow.

**Comment by amihaiemil** [comment]
@carlosmiranda Well, either way you look at it, this PR is a progress, it's moving the project forward in the direction pointed by the ticket. 

The problem is another one: the model should not be in a separate repo, since now we have an ugly dependency: ``agenda.xsd`` and ``Agenda.add(...)`` have to be modified at the same time. If we add ``title`` to the xsd and release ``datum`` before the title is actually added in ``Agenda.add(...)``, validaton will fail and vice versa. 

So, from either end you start solving this task (how I did it, or how you are suggesting), you have this dependency, the ARC has to take care of it in one task. 

Otherwise, this PR is not breaking anything and is a progress, I don't see any problem. Plus, I will drom the task if this is not accepted since I already spent the time on it, the project would lose more this way (this is not blackmailing, haha, just the real situation :D) -- this is another case of DEV does some change sand then CR/ARC would simply like a different approach, even though the proposed one is also ok... we should solve this somehow in ZC.

Let's let Yegor decide...

**Comment by amihaiemil** [comment]
@yegor256 please, have a look, you decide :D

**Comment by carlosmiranda** [comment]
@amihaiemil we have quite a few tickets here that need modifications on both repositories, and they are solved on one end while leaving a puzzle in another. So it's not as if I am suggesting a novel approach based on my personal preferences. See for example https://github.com/zerocracy/datum/pull/266/files and #515. The fact that we have separate repos is a different issue (which you've already raised in #795).

The way I see it this is not progressing the project, just making changes to what is currently irrelevant bits of code but we are restricted from making the actual change that would make it meaningful. That judgment is within my prerogative as CR. As for spending time on a task, you are aware that we (as individual DEV/CR/ARC etc.) are responsible for time management, so it's not my problem.

Not that I am being antagonistic, but that is simply how the work model works as I'm sure you are aware. I believe I am sufficiently clear. Ultimately it's @yegor256's call.

**Comment by amihaiemil** [comment]
@carlosmiranda yes, I understand your point of view. And indeed, right now, the changes done here are just "irrelevant bits of code", but it's code that will have to be written either way, so it's an improvement for the project, the ticket can be closed, CR and ARC won't be paid twice for the same review etc. 

I think in this model the first priority is to close the ticket, right? The fact that the changes done are not "productive" right away is not a problem, I would say, as long as they are in the right direction and they'll have to be done eventually.

**Comment by amihaiemil** [comment]
@carlosmiranda And yes, I usually take into account the time, I do changes in such a way that I leave 5-10min for CR changes... but nobody can forsee if the CR or ARC will require a completely different approach, that is a problem :)

**Comment by yegor256** [comment]
@amihaiemil I think I agree with @carlosmiranda -- it's better to start from datum.

As for this code, I think it contains a problem: not all jobs are GitHub tickets. We will have other jobs in the future, like JIRA jobs or Trello jobs. Your code will treat them all as GitHub jobs. I am suggesting a different design. Let's add method `title(String)` to `Agenda` and a new stakeholder `set_agenda_title_from_github.groovy`, which will get title of a job from GitHub and set it to Agenda. Make sense?

**Comment by amihaiemil** [comment]
@yegor256 @carlosmiranda I see. I'll make a PR in datum, then :)

**Comment by g4s8** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/zerocracy/farm/issues/784#issuecomment-401085636))

@g4s8 This is what I know about this job in [C3NDPUA8L](https://www.0crat.com/p/C3NDPUA8L), as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#784](https://github.com/zerocracy/farm/issues/784) is not in scope
  * The job is waiting quality review verdict by @ypshenychka[/z](https://www.0crat.com/u/ypshenychka) for 4mon
  * @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda) will receive 15 minutes after quality review
  * Job [footprint](https://www.0crat.com/footprint/C3NDPUA8L?q=%7Bjob%3A%27gh%3Azerocracy%2Ffarm%23784%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Comment by ypshenychka** [comment]
@carlosmiranda 

According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>The code reviewer found at least three problems in the code.
Comments were mostly about design problems, not cosmetic issues.

Two issues were found during code review. 
Please confirm that you'll try to find at least three major problems while future reviews.

**Comment by carlosmiranda** [comment]
@ypshenychka confirmed

**Comment by ypshenychka** [comment]
@carlosmiranda Thanks

**Comment by ypshenychka** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
Order was finished, quality is "acceptable": +15 point(s) just awarded to @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Pull request closed without merging**